### DISCUSSION TITLE
Hyper parameters tuning for Speaker Diarization

### DISCUSSION THREAD

**Discussion by SrinivasKalagotla** [pyannote/pyannote-audio#1207] [Q&A]
Hi,
I want to know how to tune the Speaker Diarization model. I saw a  piece of code from the huggingface website in which hyper parameters are instantiated. 
![image](https://user-images.githubusercontent.com/73070160/209191990-47b9414d-36f5-4ebc-b8d7-ba74a274d644.png)

I have tried the same way for speaker diarization and it throwed an error :
**ValueError: parameter 'onset' does not exist**

Can you please let me know how to tune the Speaker Diarization.

Thank you

**Comment by hbredin** [comment]
There is a [whole tutorial](https://github.com/pyannote/pyannote-audio/blob/develop/tutorials/adapting_pretrained_pipeline.ipynb) in the documentation about that.

**Reply by SrinivasKalagotla** [reply]
Thank you for the response.
I have a 120 seconds audio file with 3 speakers. I ran the Speaker Diarization on it and it is giving results with more than 120 seconds. Can you please explain why it is happening. 
![image](https://user-images.githubusercontent.com/73070160/209684819-9e7f7672-6369-4d5b-a9d3-0f4abbe059d0.png)



I think there is some issue with overlapping speech.
